### ISSUE TITLE
Debugger doesn't stop on uncaught exceptions in third party libraries

### ISSUE THREAD

**Issue by andersea** [microsoft/ptvsd#2027]
## Environment data

- VS Code version: 1.41.1
- Extension version (available under the Extensions sidebar): 2020.1.57204
- OS and version: Windows 10 Version 1909
- Python version (& distribution if applicable, e.g. Anaconda): 3.7.3 AMD 64
- Type of virtual environment used (N/A | venv | virtualenv | conda | ...): venv with poetry
- Relevant/affected Python packages and their versions: N/A
- Jedi or Language Server? (i.e. what is `"python.jediEnabled"` set to; more info microsoft/vscode-python#3977): false (language server)

Additional settings:

In the debugger tab:
I have 'Raised Exceptions' unchecked.
I have 'Uncaught Exceptions' checked.
(As a side note, I think it is very unclear how these settings are supposed to interact. I can't see anything in any documentation anywhere.)

launch.json:
```json
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python: Current File",
            "type": "python",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal"
        }
    ]
}
```

## Expected behaviour

If an exception is raised in a third party library. Code should stop on the line of code that called into the third party library.

## Actual behaviour

The exception is ignored by the debugger, causing the program to exit with an error and the exception to be printed to the terminal.

## Steps to reproduce:

(0. I tested both sync and async versions. I use trio as async library, so this needs to be added to the venv.)

1. I created a simple library with the following test functions. I used poetry with the command 'new' to create a new project, and in the project `__init__.py` file I added the following code.

`__init__.py`
```
__version__ = '0.1.2'

import trio
import time

async def aiter_thrower():
    i = 0
    while i < 5:
        yield i
        await trio.sleep(0.5)
        i += 1
    raise RuntimeError('Auch!')

async def afunc_thrower():
    await trio.sleep(0.5)
    raise RuntimeError('Auch!')

def iter_thrower():
    i = 0
    while i < 5:
        yield i
        time.sleep(0.5)
        i += 1
    raise RuntimeError('Auch!')

def func_thrower():
    time.sleep(0.5)
    raise RuntimeError('Auch!')
```

2. I build the library using poetry build. This creates a wheel file in the dist subdir.

3. I create another test project using poetry new.

4. In the test project I add the wheel file using `poetry add ..\test_library\dist\<name of wheel file>`

5. In the test project I add the following test code:

```
import trio
from async_test_lib import aiter_thrower, afunc_thrower, iter_thrower, func_thrower

async def test_afunc():
    await afunc_thrower()

async def test_aiter():
    async for x in aiter_thrower():
        print(f'Got {x}')

if __name__ == '__main__':
    print('Uncomment one of these sections in turn.')
    # Test aiter_thrower
#    trio.run(test_aiter)
    # Test afunc_thrower
#    trio.run(test_afunc)
    # Test func_thrower
#    func_thrower()
    # Test iter_thrower
#    for x in iter_thrower():
#        print(f'Got {x}')
```

6. In turn, uncomment one of the sections to test each type of raised exception.

## Logs
Output for `Python` in the `Output` panel (`View`→`Output`, change the drop-down the upper-right of the `Output` panel to `Python`)

```
(stream-trio-test-gbo-I5e8-py3.7) PS C:\Users\Anders\Documents\python\stream-trio-test> cd 'c:\Users\Anders\Documents\python\stream-trio-test'; ${env:PYTHONIOENCODING}='UTF-8'; ${env:PYTHONUNBUFFERED}='1'; & 'C:\Users\Anders\AppData\Local\pypoetry\Cache\virtualenvs\stream-trio-test-gbo-I5e8-py3.7\Scripts\python.exe' 'c:\Users\Anders\.vscode\extensions\ms-python.python-2020.1.57204\pythonFiles\ptvsd_launcher.py' '--default' '--client' 
'--host' 'localhost' '--port' '63052' 'c:\Users\Anders\Documents\python\stream-trio-test\test_asyinciter_except.py' 
Start
Got 0
Got 1
Got 2
Got 3
Got 4
Traceback (most recent call last):
  File "c:\Users\Anders\.vscode\extensions\ms-python.python-2020.1.57204\pythonFiles\ptvsd_launcher.py", line 43, in <module>
    main(ptvsdArgs)
  File "c:\Users\Anders\.vscode\extensions\ms-python.python-2020.1.57204\pythonFiles\lib\python\old_ptvsd\ptvsd\__main__.py", line 432, in main   
    run()
  File "c:\Users\Anders\.vscode\extensions\ms-python.python-2020.1.57204\pythonFiles\lib\python\old_ptvsd\ptvsd\__main__.py", line 316, in run_file
    runpy.run_path(target, run_name='__main__')
  File "C:\Users\Anders\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 263, in run_path
RuntimeError: Auch!
```

Output from `Console` under the `Developer Tools` panel (toggle Developer Tools on under `Help`; turn on source maps to make any tracebacks be useful by running `Enable source map support for extension debugging`)

```

```

**Comment by karthiknadig** [comment]
@fabioz Can you look at this one?

**Comment by fabioz** [comment]
Sure... I was actually taking a look at uncaught exceptions in third party libraries as a part #1946 (see my comment: https://github.com/microsoft/ptvsd/issues/1946#issuecomment-571611986)... I'll double check to see if this use case is also fixed when I finish #1946.

**Comment by andersea** [comment]
@fabioz @karthiknadig is this fix included in current vscode-python? Because it still doesn't work.

**Comment by int19h** [comment]
@andersea It should be included in ptvsd 5.0.0a12. You can check the version that you have with:
```py
import ptvsd
print(ptvsd.__version__)
```

**Comment by andersea** [comment]
@int19h Thanks, I had version 4, so I deleted that, made sure the above optin section was in my settings.json and now it says 5.0.0a12. The test case appears to work now. Thanks.
